cache.go

```diff
@@ -313,32 +318,75 @@ func (c *Cache[K, V]) Has(key K) bool {
 // GetOrSet returns the existing value for the key if present.
 // Otherwise, it sets and returns the given value. The retrieved
 // result is true if the value was retrieved, false if set.
+// If a loader is defined, the value won't be loaded during get.
 func (c *Cache[K, V]) GetOrSet(key K, value V, opts ...Option[K, V]) (*Item[K, V], bool) {
-	elem := c.getWithOpts(key, opts...)
+	getOpts := options[K, V]{
+		loader:            c.options.loader,
+		disableTouchOnHit: c.options.disableTouchOnHit,
+	}
+
+	c.items.mu.Lock()
+	elem := c.get(key, !getOpts.disableTouchOnHit)
```

Comment:
I like this approach but at the same time feel that the amount of repeated logic could be reduced:
1. We should probably revert back to the previous commit (i.e., no need to move half of the `get()` logic to these methods).
2. Add a special parameter to `get()` that's called `useLoader bool` or something similar and use it to determine whether the loader should be used or not (as per @davseby's suggestion)
3. Pass `true` in both `GetOrSet()` and `GetAndDelete()` but call the loader manually in the latter if the returned cache element is nil (current implementation).

Reply:
Not sure, that I got your idea.

We need the same lock to be held during get and during delete in GetOrDelete.
That means we need to do
1. lock
2. getWithOpts(useLoader=true)
3. delete
4. unlock
In this case, we stay locked during the loader execution, which we want to avoid as it can be something very long.

That's the main reason, why went with duplicating the logic from getWithOpts in the first place.

Reply:
Yes, locking should be left as is (the previous commit had this already) but there's a better way to skip the loader. We need to modify `getWithOpts()`:
```go
func (...) getWithOpts(..., useLoader bool) {
   ...
   if opts.loader != nil && useLoader {
       // call loader
   }
   ...
}
```

and `c.getWithOpts(..., true)` when calling this in `GetOrSet()` and `GetAndDelete()`.

Reply:
perfectly clear @swithek. 

Please give me some time to implement

Reply:
perfectly clear @swithek.

Please give me some time to implement

Reply:
Perfect, let me know if you nee any help @DoubleDi